### DISCUSSION TITLE
best practices for preventing multiple attempts

### DISCUSSION THREAD

**Discussion by melisa-akan** [jspsych/jsPsych#1088] [Q&A] [answered]
Hi, 

I was wondering whether there are best practices for not letting the same participants complete the experiment twice. More specifically, I would like to prevent participants from refreshing the page in the middle and restarting the experiment.

I am using the `fullscreen` plugin. I wonder if it is a good idea to alert the participants if they exit full screen mode that this action would cancel the experiment and that they would not be able to participate again.  Is there actually a way to implement such a rule?

Thanks for any and all advice!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @melisa-akan, this is something that might be easier to do on the server side. I use [JATOS](http://www.jatos.org/Whats-JATOS.html) to host my experiments, and they have some built-in options for preventing people from reloading a page and repeating a part of the experiment, or repeating the whole experiment again (for more info, see the [JATOS docs about restricting study flow](http://www.jatos.org/Restricting-study-flow.html#intro-restricting-study-flow)).

But to answer your question about what you can do about this using jsPsych: yes, you could tell people that if they exit fullscreen mode that you'll end the experiment. To set this up, I think you'd need to use the `on_interaction_data_update` parameter in the [jsPsych.init settings](https://www.jspsych.org/core_library/jspsych-core/#jspsychinit). Inside this function, you can check to see if the interaction event type was `fullscreenexit`, and if so, call the [`jsPsych.endExperiment`](https://www.jspsych.org/core_library/jspsych-core/#jspsychendexperiment) function.

Another option is to use the experiment's [`on_close` event-related callback function](https://www.jspsych.org/overview/callbacks/#on_close). Behind the scenes, jsPsych runs this function when the [`beforeunload` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) occurs, which right before the browser navigates to a different page. I think this `beforeunload` event includes the case when the page is reloaded. So to prevent someone from repeating the experiment by refreshing the page, you could add an `on_close` function that sends the data back to the server and then ends the experiment with [`jsPsych.endExperiment`](https://www.jspsych.org/core_library/jspsych-core/#jspsychendexperiment).

**Comment by melisa-akan** [comment]
Thank you so much for all the information, @becky-gilbert!
